lib/modules/onc_program/bulk_data_group_validation_sequence.rb

```diff
@@ -375,13 +375,30 @@ def streamed_ndjson_get(url, headers)
           end
         }
 
+        max_redirect = 5
+
         response = RestClient::Request.execute(
           method: :get,
           url: url,
           headers: headers,
           block_response: response_block
         )
 
+        while ['301', '302', '303', '307'].include?(response.code) && response[:location].present? && max_redirect.positive?
+          max_redirect -= 1
+          redirect_url = response[:location]
+
+          # handle relative redirects
+          redirect_url = URI.parse(url).merge(redirect_url).to_s unless redirect_url.start_with?('http')
```

Comment:
It's hard to be certain whether this works properly, so you should add a unit test for the relative redirect case.